### PR TITLE
15 FDN/J25 cards

### PR DESCRIPTION
**Pull request by Fulgur14**

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/frontline_heroism.txt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Name:Frontline Heroism
+ManaCost:2 R
+Types:Enchantment
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigToken1 | TriggerDescription$ When CARDNAME enters, create a 1/1 red Soldier creature token with haste.
+SVar:TrigToken1:DB$ Token | TokenAmount$ 1 | TokenScript$ r_1_1_soldier_haste | TokenOwner$ You
+T:Mode$ SpellCast | ValidCard$ Card | Execute$ TrigToken2 | ValidActivatingPlayer$ You | IsSingleTarget$ True | TriggerZones$ Battlefield | TargetsValid$ Creature.YouCtrl+inZoneBattlefield | TriggerDescription$ Whenever you cast a spell that targets only a single creature you control, create a 1/1 red Soldier creature token with haste, then copy that spell. The copy targets that token.
+SVar:TrigToken2:DB$ Token | TokenAmount$ 1 | TokenScript$ r_1_1_soldier_haste | TokenOwner$ You | RememberTokens$ True | SubAbility$ DBCopy
+SVar:DBCopy:DB$ CopySpellAbility | Defined$ TriggeredSpellAbility | Controller$ You | CopyForEachCanTarget$ Permanent.IsRemembered | TriggeredSpellAbilityTargetsOrController | ChooseOnlyOne$ True | SubAbility$ DBCleanup
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/frontline_heroism.txt:8]: bad param

File: forge-gui/res/cardsfolder/upcoming/generous_pup.txt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+Name:Generous Pup
+ManaCost:1 W
+Types:Creature Dog
+PT:2/2
+K:Vigilance
+T:Mode$ CounterAddedOnce | ValidCard$ Card.Self | TriggerZones$ Battlefield | OptionalDecider$ You | CounterType$ P1P1 | Execute$ TrigPutCounterAll | ActivationLimit$ 1 | TriggerDescription$ Whenever one or more +1/+1 counters are put on Generous Pup, put a +1/+1 counter on each other creature you control. This ability triggers only once each turn.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/generous_pup.txt:6]: CARDNAME

Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/generous_pup.txt:6]: not optional

File: forge-gui/res/cardsfolder/upcoming/niv_mizzet_visionary.txt [hunk 3]
```diff
@@ -0,0 +1,11 @@
+Name:Niv-Mizzet, Visionary
+ManaCost:4 U R
+Types:Legendary Creature Dragon Wizard
+PT:5/5
+K:Flying
+S:Mode$ Continuous | Affected$ You | SetMaxHandSize$ Unlimited | Description$ You have no maximum hand size.
+T:Mode$ DamageDone | ValidSource$ Card.YouCtrl,Emblem.YouCtrl | ValidTarget$ Opponent | CombatDamage$ False | TriggerZones$ Battlefield | Execute$ TrigDraw | TriggerDescription$ Whenever a source you control deals noncombat damage to an opponent, you draw that many cards.
+SVar:TrigDraw:DB$ Draw | NumCards$ X
+SVar:X:TriggerCount$DamageAmount
+DeckHints:Ability$Token
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/niv_mizzet_visionary.txt:10]: wrong hint?

File: forge-gui/res/cardsfolder/upcoming/taeko_the_patient_avalanche.txt [hunk 4]
```diff
@@ -0,0 +1,15 @@
+Name:Taeko, the Patient Avalanche
+ManaCost:3 U
+Types:Legendary Creature Turtle Ninja
+PT:4/5
+R:Event$ Moved | ValidCard$ Card.Self | Destination$ Battlefield | ReplacementResult$ Updated | ReplaceWith$ ETBTapped | Description$ CARDNAME enters tapped.
+SVar:ETBTapped:DB$ Tap | Defined$ Self | ETB$ True
+T:Mode$ ChangesZone | Origin$ Battlefield | Destination$ Ante,Command,Exile,Hand,Library | ValidCard$ Creature.Other+YouCtrl | TriggerZones$ Battlefield | Execute$ TrigScry | TriggerDescription$ \nWhenever another creature you control leaves the battlefield, if it didn't die, scry 1 and put a +1/+1 counter on NICKNAME.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/taeko_the_patient_avalanche.txt:7]: bad description

File: forge-gui/res/cardsfolder/upcoming/frontline_heroism.txt (hunk 1 above)
Comment (Fulgur14) [reply forge-gui/res/cardsfolder/upcoming/frontline_heroism.txt:8]: It's probably a bit too late here for me -- which param?

Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/frontline_heroism.txt:8]: "TriggeredSpellAbilityTargetsOrController"?

File: forge-gui/res/cardsfolder/upcoming/tinybones_bauble_burglar.txt [hunk 5]
```diff
@@ -0,0 +1,9 @@
+Name:Tinybones, Bauble Burglar
+ManaCost:1 B
+Types:Legendary Creature Skeleton Rogue
+PT:1/3
+T:Mode$ Discarded | ValidCard$ Card.OppOwn | TriggerZones$ Battlefield | Execute$ TrigExile | TriggerDescription$ Whenever an opponent discards a card, exile it from their graveyard with a stash counter on it.
+SVar:TrigExile:DB$ ChangeZone | Defined$ TriggeredCard | Origin$ Graveyard | Destination$ Exile | WithCountersType$ STASH
+S:Mode$ Continuous | Condition$ PlayerTurn | MayPlay$ True | Affected$ Card.YouDontOwn+counters_GE1_STASH | AffectedZone$ Exile | MayPlayIgnoreType$ True | Description$ During your turn, you may play cards you don't own with stash counters on them from exile, and mana of any type can be spent to cast those spells.
+A:AB$ Discard | Cost$ 3 B T | Defined$ Opponent | Mode$ TgtChoose | SorcerySpeed$ True | SpellDescription$ Each opponent discards a card.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/tinybones_bauble_burglar.txt:8]: description incomplete

**Review by tool4ever — APPROVED** [review]

**Pull request merged**